crates/sui-grpc-rosetta/src/account.rs

```diff
@@ -153,56 +174,42 @@ async fn get_account_balances(
 
 async fn get_sub_account_balances(
     account_type: SubAccountType,
-    client: &SuiClient,
+    grpc_client: &mut sui_rpc::client::Client,
     address: SuiAddress,
 ) -> Result<Vec<Amount>, Error> {
-    let amounts = match account_type {
-        SubAccountType::Stake => {
-            let delegations = client.governance_api().get_stakes(address).await?;
-            delegations.into_iter().fold(vec![], |mut amounts, stakes| {
-                for stake in &stakes.stakes {
-                    if let StakeStatus::Active { .. } = stake.status {
-                        amounts.push(SubBalance {
-                            stake_id: stake.staked_sui_id,
-                            validator: stakes.validator_address,
-                            value: stake.principal as i128,
-                        });
-                    }
-                }
-                amounts
+    let current_epoch = get_current_epoch(grpc_client).await?;
+    let address = Address::from(address);
+    let delegated_stakes = grpc_client.list_delegated_stake(&address).await?;
+
+    let amounts: Vec<SubBalance> = match account_type {
+        SubAccountType::Stake => delegated_stakes
+            .into_iter()
+            .filter(|stake| current_epoch >= stake.activation_epoch)
+            .map(|stake| SubBalance {
```

Comment:
you could change the SubBalance type to use types from sui-sdk-types instead of sui-types to avoid the conversions.